### PR TITLE
Eliminate literal parameters to uvector::set_element_async and device_scalar::set_value

### PR DESCRIPTION
**Pull request by harrism**
After rapidsai/rmm#725 is merged, this PR updates cuspatial to eliminate passing literal values to `device_scalar::set_value`.

Note there are still similar issues with uses of `cudf::scalar` that need to be addressed. But I wanted to get this open before the end of the week. This can be merged after RMM 725 is merged to fix the build, so I think the `scalar` issues should be fixed separately.

### CODE REVIEW

File: cpp/tests/jit/jit-cache-multiprocess-test.cpp [hunk 1]
```diff
@@ -49,8 +49,10 @@ TEST_F(JitCacheMultiProcessTest, MultiProcessTest)
     // Brand new cache object that has nothing in in-memory cache
     cudf::jit::cudfJitCache cache;
 
-    input->set_value(4);
-    output->set_value(1);
+    auto const in{4};
+    auto const out{1};
+    input->set_value(in);
+    output->set_value(out);
```
Comment (karthikeyann) [inline cpp/tests/jit/jit-cache-multiprocess-test.cpp:55]: why this change?

Comment (harrism) [reply cpp/tests/jit/jit-cache-multiprocess-test.cpp:55]: https://github.com/rapidsai/rmm/pull/725

**Comment by harrism** [comment]
This will not pass CI until rapidsai/rmm#725 is merged.

**Comment by harrism** [comment]
rerun tests

**Review by devavret — APPROVED** [review]

**Review by trxcllnt — APPROVED** [review]

**Comment by galipremsagar** [comment]
rerun tests

**Comment by harrism** [comment]
@gpucibot merge

**Comment by galipremsagar** [comment]
rerun tests
